### ISSUE TITLE
File downloads on chrome extension startup

### ISSUE THREAD

**Issue by vinhdang84** [vinhdang84/campaign-data-downloader#2]
When I reload or start chrome, the chrome extension loads. On load, the file  gets downloaded. I want the file to only be downloaded in either of these two paths. The second being preferred. 
(a) I click on the extension 
(b) or on button click within the popup.html

Right now the extension downloads the file on the extension reloading and on scenario (a) from above.

**Comment by vinhdang84** [comment]
To do scenario (b) which is to have a button in popup.html and have the script run on button click, I would need to create a button within popup.html, which has an ID. Within the popup.js, I would have an event listener to listen for that ID and will then run the getbackgroundpage and execute the function in the background.js. 


"But you can always add custom eventListeners on background.js scripts which you are trigger from content scripts using the messaging feature. A simple example is depicted in the documentation. Where you send a custom message from content script and receive the message on background script. This way you can communicate with background script."

https://developer.chrome.com/extensions/messaging#simple
-https://stackoverflow.com/questions/45017841/call-event-listener-from-background-js-in-content-script

**Comment by vinhdang84** [comment]
I would need to do the following: 

1. Add a button to background.html. 
2. Give button an ID.
3. Add event listener for this ID in content.js. This event listener will run the function from background.js
4. Background.js should have another event listener. 

I think steps 3 and 4 seem redundant. Do I need to run two event listeners? The tricky part is how I can't run this download function within content.js. However, I feel this could be more efficient. For now I'll run it this way. 

I think another tricky part is how the content.js is commanded to run the background.js function. It seems like I'll need to add another event ID here to have background.js to work. I could spend some time researching first but could be good to do it incorrectly anyways to get more practice with event messaging.

**Comment by vinhdang84** [comment]
* I changed background.js filename to download.js and kept it as a background script in manifest. 
* I changed popup.js to main.js. Now it contains a function to run download.js function using the getBackgroundPage method. More importantly, I added an event listener that listens for an event from popup.html... a button click that corresponds to id 'downloadfile'
* I changed popup.html to main.html. Main.html now contains a button with element ID 'downloadfile'. It also loads main.js so that this file can listen for the button press. 


The script now downloads the csv file on button press. It doesn't run the function 'downloadFile' within download.js on startup anymore. I don't understand why this characteristic changed although it is what I want. Issue needs to be kept open until resolved.
